nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/test/java/org/apache/nifi/snmp/processors/ListenTrapSNMPTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.snmp.processors;
+
+import org.apache.nifi.remote.io.socket.NetworkUtils;
+import org.apache.nifi.snmp.context.SNMPClientFactory;
+import org.apache.nifi.snmp.helper.SNMPTestUtils;
+import org.apache.nifi.snmp.operations.SNMPTrapSender;
+import org.apache.nifi.snmp.utils.SNMPUtils;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Test;
+import org.snmp4j.CommunityTarget;
+import org.snmp4j.PDUv1;
+import org.snmp4j.Snmp;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.TimeTicks;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ListenTrapSNMPTest {
+
+    @Test
+    public void testReceive() throws InterruptedException {
+
+        int port = NetworkUtils.availablePort();
+        CommunityTarget target = SNMPTestUtils.createCommTarget("public", "127.0.0.1/" + port, SnmpConstants.version2c);
+
+        TestRunner runner = getTestRunner(port);
+        runner.run(1, false);
+        Thread.sleep(200);
+
+        String enterpriseOID = "1.3.6.1.4.1.1824";
+        String agentAddress = "1.2.3.4";
+        int genericTrapType = PDUv1.ENTERPRISE_SPECIFIC;
+        int specificTrapType = 2;
+        OID trapOID = new OID("1.3.6.1.4.1.1234.2.1.51");
+        String managerAddress = "1.2.3.5";
+        String trapOIDValue = "TrapOidValue";
+        TimeTicks sysUpTime = new TimeTicks(5000);
+
+        Snmp snmp = SNMPClientFactory.createSnmpClient(String.valueOf(NetworkUtils.availablePort()));
+
+        try {
+            SNMPTrapSender trapSender = new SNMPTrapSender(snmp, target);
+            trapSender.generateTrap(sysUpTime, enterpriseOID, agentAddress, genericTrapType, specificTrapType, trapOID, managerAddress, trapOIDValue);
+            Thread.sleep(200);
```

Comment:
Why this is necessary?

Reply:
SNMP requests are asynchronous. I checked and 20 ms is enough for the message to arrive. I found awaitility complicated at first glance and this is the only place where a little stop is needed.